protocols/direct-funding_test.go

```diff
@@ -1,14 +1,88 @@
 package protocols
 
 import (
+	"math/big"
 	"testing"
 
+	"github.com/ethereum/go-ethereum/common"
 	"github.com/statechannels/go-nitro/channel/state"
+	"github.com/statechannels/go-nitro/types"
 )
 
+// TestNew tests the constructor using a TestState fixture
 func TestNew(t *testing.T) {
 	_, err := NewDirectFundingObjectiveState(state.TestState, state.TestState.Participants[0])
 	if err != nil {
 		t.Error(err)
 	}
 }
+
+// Construct various variables for use in TestUpdate
+var s, _ = NewDirectFundingObjectiveState(state.TestState, state.TestState.Participants[0])
+var dummySignature = state.Signature{
+	R: common.Hex2Bytes(`49d8e91bd182fb4d489bb2d76a6735d494d5bea24e4b51dd95c9d219293312d9`),
+	S: common.Hex2Bytes(`22274a3cec23c31e0c073b3c071cf6e0c21260b0d292a10e6a04257a2d8e87fa`),
+	V: byte(1),
+}
+var dummyStateHash = common.Hash{}
+var stateToSign state.State = s.ExpectedStates[0]
+var stateHash, _ = stateToSign.Hash()
+var privateKeyOfParticipant0 = common.Hex2Bytes(`caab404f975b4620747174a75f08d98b4e5a7053b691b41bcfc0d839d48b7634`)
+var correctSignatureByParticipant, _ = stateToSign.Sign(privateKeyOfParticipant0)
+
+func TestUpdate(t *testing.T) {
+
```

Comment:
This test looks like it is essentially constructing various events and sending them to the update method on an initial test state, and asserting expected results. A common pattern for golang for this is table tests -- https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

The above blog lays out a good explanation. It can take a tiny bit of getting used to but I find myself using them a lot.

Reply:
We love table tests: https://github.com/statechannels/statechannels/search?q=test.each

Reply:
We do, and we've now started writing them in Go: https://github.com/statechannels/go-nitro/pull/86/commits/859a1eb635beba8a8677d61cb7e94576f2ae43c8 .